src/intervals/interval_rounding.jl

```diff
@@ -0,0 +1,193 @@
+""" ->
+
+get_interval_rounding() = INTERVAL_ROUNDING[end]
+
+set_interval_rounding(mode) = INTERVAL_ROUNDING[end] = mode  # a symbol
+
+
+
+set_interval_rounding(:narrow)
+
+
+macro with_rounding(T, expr, rounding_mode)
+    quote
+        with_rounding($T, $rounding_mode) do
+            $expr
+        end
+    end
+end
+
+
+@doc doc"""The `@round` macro creates a rounded interval according to the current interval rounding mode.
+It is the main function used to create intervals in the library (e.g. when adding two intervals, etc.).
+It uses the interval rounding mode (see get_interval_rounding())""" ->
+macro round(T, expr1, expr2)
+    #@show "round", expr1, expr2
+    quote
+        mode = get_interval_rounding()
+
+        if mode == :wide  #works with any rounding mode set, but the result will depend on the rounding mode
+            # we assume RoundNearest
+            Interval(prevfloat($expr1), nextfloat($expr2))
+
+        else # mode == :narrow
+            Interval(@with_rounding($T, $expr1, RoundDown), @with_rounding($T, $expr2, RoundUp))
+        end
+
+    end
+end
+
+@doc doc"""`@thin_round` possibly saves one operation compared to `@round`.""" ->
+macro thin_round(T, expr)
+    quote
+        mode = get_interval_rounding()
+
+        if mode == :wide  #works with any rounding mode set, but the result will depend on the rounding mode
+            # we assume RoundNearest
+            temp = $expr  # evaluate the expression
+            Interval(prevfloat(temp), nextfloat(temp))
+
+        else # mode == :narrow
+            Interval(@with_rounding($T, $expr, RoundDown), @with_rounding($T, $expr, RoundUp))
+
+        end
+    end
+end
+
+
+@doc doc"""`make_interval` is used by `@interval` to create intervals from individual elements of different types"""->
+make_interval(::Type{BigFloat}, x::String)    =  @thin_round(BigFloat, @compat parse(BigFloat,x))
+
+make_interval(::Type{BigFloat}, x::String)    =  @thin_round(BigFloat, BigFloat(x))
+make_interval(::Type{BigFloat}, x::MathConst) =  @thin_round(BigFloat, big(x))
+
+make_interval(::Type{BigFloat}, x::Integer)   =  Interval(BigFloat(x))  # no rounding -- dangerous if very big integer
+# but conversion from BigInt to BigFloat with correct rounding seems to be broken anyway # @thin_interval(BigFloat("$x"))
+
+make_interval(::Type{BigFloat}, x::Rational)  =  make_interval(BigFloat, x.num) / make_interval(BigFloat, x.den)
+make_interval(::Type{BigFloat}, x::Float64)   =  make_interval(BigFloat, rationalize(x))  # NB: converts a float to a rational -- this could be slow
+
+make_interval(::Type{BigFloat}, x::BigInt)    =  @thin_round(BigFloat, convert(BigFloat, x))
+
+make_interval(::Type{BigFloat}, x::BigFloat)  =  @thin_round(BigFloat, 1.*x)  # convert to possibly different BigFloat precision
+make_interval(::Type{BigFloat}, x::Interval)  =  @round(BigFloat, big(1.*x.lo), big(1.*x.hi))
+
+
+
+make_interval(::Type{Float64}, x::String)    =  @thin_round(Float64, parsefloat(x))
```

Comment:
`@floatinterval("0.1")` triggers a deprecation warning to use `parse(Float64,s)`.